tests/fourier/test_fourier_utils.py

```diff
@@ -12,12 +12,69 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-Unit tests for :mod:`fourier` utils functions.
+Tests for the Fourier module helper functions.
 """
-from pennylane.fourier.utils import format_nvec
+import pytest
+import numpy as np
+import pennylane as qml
+from pennylane.fourier.utils import (
+    format_nvec,
+    get_spectrum,
+    join_spectra,
+)
 
 
-def test_format_nvec():
-    """Test the format_nvec function."""
-    assert format_nvec(1) == str(1)
-    assert format_nvec([1, -1]) == " 1 -1"
+@pytest.mark.parametrize(
+    "nvec, exp",
+    [(1, "1"), (-20, "-20"), ((23,), " 23"), ((-1,), "-1"), ((2, -1, 42), " 2 -1  42")],
+)
+def test_format_nvec(nvec, exp):
+    """Test formatting of a tuple of integers into a nice string."""
+    assert format_nvec(nvec) == exp
+
+
+@pytest.mark.parametrize(
+    "spectrum1, spectrum2, expected",
+    [
+        ({0, 1}, {0, 1}, {0, 1, 2}),
+        ({0, 3}, {0, 5}, {0, 2, 3, 5, 8}),
+        ({0, 1, 2}, {0, 1}, {0, 1, 2, 3}),
+        ({0, 0.5}, {0, 1}, {0, 0.5, 1.0, 1.5}),
+        ({0, 0.5}, {0}, {0, 0.5}),
+        ({0}, {0, 0.5}, {0, 0.5}),
+    ],
+)
+def test_join_spectra(spectrum1, spectrum2, expected, tol):
+    """Test that spectra are joined correctly."""
+    joined = join_spectra(spectrum1, spectrum2)
+    assert joined == expected
+    # assert np.allclose(sorted(joined), expected, atol=tol, rtol=0)
```

Comment:
Why is this check commented out? If it's not needed, it can be removed and the `tol` argument can also be removed from the function header.

Reply:
Sorry, forgot to remove!